### ISSUE TITLE
Leaf lighting is often stale

### ISSUE THREAD

**Issue by LogicParrot** [cuberite/cuberite#3103]
When a leaf is destroyed, the lighting values are not updated properly until a seemingly random nearby block is destroyed.
Potentially related / dupe of: #133 

Video: https://vimeo.com/160752349

**Comment by bearbin** [comment]
I have been working on this issue for a little bit, and I can't work out where, after the chunk has its lighting marked dirty, the chunk is added to the relighting queue.

Could anybody help me understand the code - @madmaxoft possibly?

**Comment by tigerw** [comment]
I don't think we relight, unless something needs to lighting data. For example, grass explicitly requests lighting if it finds the light is dirty.

Possibly chunks are relit when they are loaded too.
